primitives/src/v2/mod.rs

```diff
@@ -1569,6 +1594,72 @@ impl CompactStatement {
 	}
 }
 
+/// `IndexedVec` struct indexed by type specific indexes.
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(PartialEq, MallocSizeOf))]
+pub struct IndexedVec<K, V>(Vec<V>, PhantomData<fn(K) -> K>);
+
+impl<K, V> Default for IndexedVec<K, V> {
+	fn default() -> Self {
+		Self(vec![], PhantomData)
+	}
+}
+
+impl<K, V> From<Vec<V>> for IndexedVec<K, V> {
+	fn from(validators: Vec<V>) -> Self {
+		Self(validators, PhantomData)
+	}
+}
+
+impl<K, V> FromIterator<V> for IndexedVec<K, V> {
+	fn from_iter<T: IntoIterator<Item = V>>(iter: T) -> Self {
+		Self(Vec::from_iter(iter), PhantomData)
+	}
+}
+
+impl<K, V> IndexedVec<K, V>
+where
+	V: Clone,
+{
+	/// Returns a reference to an element indexed using `K`.
+	pub fn get(&self, index: K) -> Option<&V>
+	where
+		K: Into<u32>,
+	{
+		self.0.get(index.into() as usize)
+	}
+
+	/// Returns number of elements in vector.
+	pub fn len(&self) -> usize {
+		self.0.len()
+	}
+
+	/// Returns contained vector.
+	pub fn to_vec(&self) -> Vec<V> {
+		self.0.clone()
+	}
+
+	/// Returns Iterator of contained vector.
+	pub fn iter(&self) -> Iter<'_, V> {
+		self.0.iter()
+	}
+
+	/// Returns a mutable Iterator.
+	pub fn iter_mut(&mut self) -> IterMut<'_, V> {
+		self.0.iter_mut()
+	}
```

Comment:
Is it used anywhere? If not, I'd suggest deleting.

Reply:
Strange, thought it was used somewhere. Will remove

Reply:
It's used `node/network/availability-distribution/src/requester/session_cache.rs` line 195